File: coodingpenguin/programmers/level2/전화번호목록.py
def solution(phone_book):
    phone_book.sort()
    for i in range(len(phone_book)-1):
        # 바로 뒤에 전화번호의 접두어가 앞의 전화번호인지 확인
        if phone_book[i+1].startswith(phone_book[i]):
            return False
    return True

File: sally/programmers/week6/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    
    int size_ = phone_book.size();
    sort(phone_book.begin(), phone_book.end());
    
    for (int i = 0; i < size_-1; i++) {
        int s = phone_book[i].size();
        if (phone_book[i].compare(phone_book[i+1].substr(0, s)) == 0) 
            return false;
    }

    return answer;
}